editor/src/messages/portfolio/document/utility_types/network_interface.rs

```diff
@@ -6212,118 +6512,92 @@ pub enum WidgetOverride {
 }
 
 // TODO: Custom deserialization/serialization to ensure number of properties row matches number of node inputs
-#[derive(Debug, Clone, PartialEq, serde::Serialize, serde::Deserialize)]
-pub struct PropertiesRow {
+#[derive(Debug, Clone, Default, PartialEq, serde::Serialize, serde::Deserialize)]
+pub struct InputPersistentMetadata {
 	/// A general datastore than can store key value pairs of any types for any input
-	// TODO: This could be simplified to just Value, and key value pairs could be stored as the Value::Object variant
+	/// Each instance of the input node needs to store its own data, since it can lose the reference to its
+	/// node definition if the node signature is modified by the user. For example adding/removing/renaming an import/export of a network node.
 	pub input_data: HashMap<String, Value>,
 	// An input can override a widget, which would otherwise be automatically generated from the type
 	// The string is the identifier to the widget override function stored in INPUT_OVERRIDES
 	pub widget_override: Option<String>,
-	#[serde(skip)]
+	/// An empty input name means to use the type as the name.
 	pub input_name: String,
-	#[serde(skip)]
+	/// Displayed as the tooltip.
 	pub input_description: String,
 }
 
-impl Default for PropertiesRow {
-	fn default() -> Self {
-		("", "TODO").into()
-	}
```

Comment:
The "TODO" got replaces with "" when sending to the frontend, so theres no reason to default to it